include/adapters/script_runner.hpp

```diff
@@ -12,8 +12,8 @@ POLYBAR_NS
 class script_runner {
  public:
   using interval = std::chrono::duration<double>;
-  script_runner(std::function<void(void)> on_update, const string& exec, const string& exec_if, bool tail,
-      interval interval, const vector<pair<string, string>>& env);
+  script_runner(std::function<void(void)> on_update, string exec, string exec_if, bool tail, interval interval,
```

Comment:
I don't agree with using rvalues here. You don't gain anything since you still have a copy (now it happens when calling the constructor instead of during the constructor).

Even when using rvalue references, from a performance standpoint, the `script_runner` is only executed when the module starts up and the two strings are generally no more than 100 bytes.

Unless you there is a good reason for this, I'd rather keep using const references.

Reply:
Well, i think it's kind of a personal preference. Passing an argument by-value offers clearer function interface by explicitly stating that argument will be copied when passed to the function. Const reference, on the other hand, only _suggests_, that it will and doesn't really specify the nature of interaction with the argument.

Performance-wise, when copy of the argument is desired inside a function, passing by value and then moving is better than passing by const reference, since if rvalue is passed, binding to const reference will always copy instead of just moving. Although, as you noted, it doesn't really matter in this case.

```c++
void f1(std::string s);
void f2(const std::string s);

auto s = std::string();
f1(std::move(s)); // move
f2(std::move(s)); // copy
```

The reason I made this change is because I tend to generally follow this principle, mostly because of clearer interface, than performance reasons. Didn't think much there, just an automatic reaction. I'll roll this one back to const references.

Reply:
Thanks for the explanation :)

I guess it really is a personal preferences. I would rather not expose the fact that a copy is made in the interface because if the copy becomes unnecessary in the future, it changes the interface and you may have to update all users.